### DISCUSSION TITLE
Binder Design

### DISCUSSION THREAD

**Discussion by zarbuk** [RosettaCommons/RFdiffusion#218] [Q&A] [answered]
Hi, I am trying to design binders for a specific enzyme, I am using a specific scaffold and the length is 106 aa. If I want to design the binder to be 55 aa, how do I do that? 
I tried using contig length however, it does not work like that.

**Comment by roccomoretti** [comment]
Your use case sounds almost identical to the standard Binder Design use case outlined in the [README](https://github.com/RosettaCommons/RFdiffusion?tab=readme-ov-file#binder-design) and the examples. I'd recommend working through the provided examples, then try to mimic that setup with your own system.

If that isn't working for you, you may need to provide more details on what you tried and how "not working" expresses itself in your particular case.

**Comment by zarbuk** [comment]
Hello,

Thank you for the inputs. The example provided talks about lengthening the
tim barrels. In my case I need to shorten the length of my binder . In
order to do that i used contig lenth command and the output is still 106 aa
long based on the scaffold lenth. I am also interested in positioning this
binder to a certain area over the enzyme.

Should i post the code here?

**Comment by roccomoretti** [comment]
The examples in the "Binder Design" and "Practical Considerations for Binder Design" don't make any mention of TIM barrels. The https://github.com/RosettaCommons/RFdiffusion/blob/main/examples/design_ppi.sh example uses the insulin receptor  (The "Fold Conditioning" section does mention TIM barrels, but that's a completely separate protocol.)

If that isn't helping, then posting details of what you want to accomplish, the best-effort command line you attempted to use, and why that command didn't work for you would help with providing guidance.

**Comment by zarbuk** [comment]
Hi ,

I will break down my objective: I want to design a binder to an enzyme at a
specific site and it should be approximately 55 aa long and should fold
into a three helical bundle  (for now, I would not want to try
other topologies) . I picked 3FOO PDB file which is a 106 aa long helical
bundle.

Here is my command that I used :
../scripts/run_inference.py
scaffoldguided.target_path=/path/to/corrected.pdb
inference.output_prefix=/output/path/design1
scaffoldguided.scaffoldguided=True 'ppi.hotspot_res=[B6,B10,B66,B96,B111]'
scaffoldguided.target_pdb=True scaffoldguided.target_pdb=True
scaffoldguided.target_ss=/home/zbukhari/RFdiffusion/ppi_scaffold_targetfold/
target_ss.pt
scaffoldguided.target_adj=/home/zbukhari/RFdiffusion/ppi_scaffold_targetfold/
target_adj.pt
scaffoldguided.scaffold_dir=/path/to/templatedirectory/template
inference.num_designs=10000 denoiser.noise_scale_ca=0.0
denoiser.noise_scale_frame=0.0 contigmap.length=55-55
scaffoldguided.mask_loops=False

The output that I got is of 106 aa because my template was 106 aa long.
I also had a question regarding setting the denoiser scale.ca to 0 instead
of 0.5 , will that impact a lot?

Typically, if I had to make my design highly specific to an enzyme that
shares 99% homology, how do you recommend I should go about it, what is the
comparison between Rosetta Two state designs and RF diffusion? Any
suggestion in this direction would also help me.

Regards
Zara

**Reply by roccomoretti** [reply, marked as answer]
I'm personally not too familiar with the scale.ca -- I might recommend posing that as a separate question.

Helical bundles should be rather simple for RFDiffusion to generate. You should be able to use the [Fold Conditioning](https://github.com/RosettaCommons/RFdiffusion?tab=readme-ov-file#fold-conditioning) approach to specify your desired topology. 

Note that your specified topology should be your _desired_ topology. In your case where you have a 106 aa template structure but only want a 55 aa protein, you're going to have to trim down the template structure. Manually (e.g. with a text editor or with the structure altering tools of PyMol or Chimera) remove all the residues from the template structure which don't match the final topology you're interested in. Once you have all that trimmed down, I believe you should be able to use that to make your secondary structure adjacency file. (Keep in mind that there's nothing special about the adjacency files that they *need* to be generated by the script. You can manually put them together (or edit them) if you know the syntax.)

There's an example specifically for binder design with fold conditioning at `examples/design_ppi_scaffolded.sh`

**Comment by zarbuk** [comment]
I was doing that ( Using fold conditions and the binder design option
there) .Thank you so much for the detailed answer . I actually ran a DALI
to get various other similar structures and then selected  other options
with varying length. That ways I had a folder of scaffolds and topology
that I could use.

I dint quite understand the concept of scale.ca, Ill take your
recommendation and post that as another question.

Thank you for the help.
